packages/xlucene-evaluator/src/parser/functions/geo/polygon.ts

```diff
@@ -0,0 +1,63 @@
+
+// @ts-ignore
+import { polygon, lineString } from '@turf/helpers';
+// @ts-ignore
+import lineToPolygon from '@turf/line-to-polygon';
+import { geoMatcher } from './helpers';
+import { parseGeoPoint } from '../../../utils';
+import * as i from '../../interfaces';
+import { AnyQuery } from '../../../translator';
+
+function validate(params: i.Term[]) {
+    const geoPointsParam = params.find((node) => node.field === 'points');
+    if (geoPointsParam == null) throw new Error('geoPolygon query needs to specify a "points" parameter');
+    if (!Array.isArray(geoPointsParam.value)) throw new Error('points parameter must be an array');
+    // @ts-ignore we are ignoreing util we have a better story around list expressions
+    const points = geoPointsParam.value.map((node) => {
+        if (!node.value) throw new Error('points parameter must be an array of string values');
+        return parseGeoPoint(node.value);
+    });
+
+    if (points.length < 3) throw new Error('geoPolygon points parameter must have at least three points');
+
+    return points;
+}
+
+const geoPolygon: i.FunctionDefinition = {
+    name: 'geoPolygon',
+    version: '1',
+    create(field: string, params: any, { logger }) {
+        if (!field || field === '*') throw new Error('field for geoBox cannot be empty or "*"');
```

Comment:
Make sure you to double check your copy and pasted code:

```suggestion
        if (!field || field === '*') throw new Error('field for geoPolygon cannot be empty or "*"');
```

Reply:
fixed